### PR TITLE
Admin/team billing downgrader

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Migrates @jamiepine downgraded to a self contained script 
- Fixes some prisma conflicts with `ts-node`
- Updates trial logic accordingly

### CODE REVIEW

**Review by alannnc — COMMENTED** [review]
Is this script meant to be run several times or just once?

**Review by alannnc — APPROVED** [review]

**Comment by zomars** [comment]
> Is this script meant to be run several times or just once?

Just once

File: apps/web/ee/lib/stripe/downgrade.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        plan: {
+          not: UserPlan.PRO,
+        },
+      },
+    },
+    include: {
+      user: true,
+    },
+  });
+  const downgrade = async (member: typeof illegalProUsers[number]) => {
```
[the remark this answers is not in the document]
Comment (PeerRich) [reply apps/web/ee/lib/stripe/downgrade.ts:28]: dont worry this will only ne executed once

File: apps/web/@prisma/client.ts [hunk 2]
```diff
@@ -1 +0,0 @@
-export * from "@calcom/prisma/client";
```
Comment (zomars) [inline apps/web/@prisma/client.ts:1]: This file got deleted for a conflict with ts-node

File: apps/web/ee/components/TrialBanner.tsx [hunk 3]
```diff
@@ -13,9 +13,11 @@ const TrialBanner = () => {
...
   if (!user || user.plan !== "TRIAL") return null;
 
-  const trialDaysLeft = dayjs(user.createdDate)
-    .add(TRIAL_LIMIT_DAYS + 1, "day")
-    .diff(dayjs(), "day");
+  const trialDaysLeft = user.trialEndsAt
+    ? dayjs(user.trialEndsAt).add(1, "day").diff(dayjs(), "day")
+    : dayjs(user.createdDate)
+        .add(TRIAL_LIMIT_DAYS + 1, "day")
+        .diff(dayjs(), "day");
```
Comment (zomars) [inline apps/web/ee/components/TrialBanner.tsx:20]: Here I did this allow us to extend trials for non-recently created users.

**Review by zomars — COMMENTED** [review]
Ready for review

File: apps/web/ee/lib/stripe/downgrade.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    // if their pro is already sponsored by a team, do not downgrade
+    if (metadata.proPaidForTeamId !== undefined) continue;
+
+    const stripeCustomerId = await getStripeCustomerIdFromUserId(member.user.id);
+    if (!stripeCustomerId) {
+      await downgrade(member);
+      continue;
+    }
+
+    const customer = await stripe.customers.retrieve(stripeCustomerId, {
```
Comment (alannnc) [inline apps/web/ee/lib/stripe/downgrade.ts:54]: @zomars I don't know the size of the current illegal pro users, but just keep in mind stripe api rate limit here?
100 reads per second
https://stripe.com/docs/rate-limits

**Comment by agustif** [comment]
The code looks good to me!

The only idea I can think of if we're missing a lot of numbers of what we're expecting. is to invert this into an upgradeScript that checks for presence of current subscription or other status and then upgrades the user? Maybe inverting the baseline to everyone is on trial and we update the ones who we know are paid users works better?

**Pull request merged**